### PR TITLE
Math-Magician : Events

### PR DESCRIPTION
**Pull request by bhushan354**
In this PR I have implemented following changes : 

- Implemented click event handlers.
- Used big.js library with Components.
- Used useState and props.
- Added styling .

Thank you ,
Bhushan .

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  if (!inObject.newNum && inObject.sum && !inObject.action) {
+    return { ...inObject, action: numName };
+  }
+
+  return {
+    sum: inObject.newNum,
+    action: numName,
+    newNum: null,
+  };
+}
```
Comment (Johnadibe) [inline src/logic/calculate.js]: > - [ ] Hello @bhushan354, I noticed that you made modifications to the calculate.js file provided by Microverse. It's recommended to use the calculate.js file exactly as it was provided by Microverse without any alterations. If you have made any changes to this file, please revert those changes to ensure it remains in its original state. This approach is essential for maintaining consistency with the project requirements and adhering to best practices.👍
> 
> Kindly utilize the version provided by Microverse, accessible [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js).

File: src/logic/Operate.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      try {
+        return firstNum.mod(secondNum).toString();
+      } catch (err) {
+        return "Can't find modulo as can't divide by 0.";
+      }
+
+    default:
+      throw Error(`Unknown operatorValue '${operatorValue}'`);
+  }
+}
```
Comment (Johnadibe) [inline src/logic/Operate.js]: > - [ ] Hello @bhushan354, I noticed that you made modifications to the operate.js file provided by Microverse. It's recommended to use the operate.js file exactly as it was provided by Microverse without any alterations. If you have made any changes to this file, please revert those changes to ensure it remains in its original state. This approach is essential for maintaining consistency with the project requirements and adhering to best practices.👍
> 
> Kindly utilize the version provided by Microverse, accessible [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js).👍

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @bhushan354👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Linter checks passed✔️
✅ Created a new directory called logic✔
✅ App works as expected✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bhushan354** [comment]
hii sir/madam , i have setup project aa you said  and also added a child file to my components .

Thank you ,
Bhushan .

**Review by algerina — APPROVED** [review]
Hi @bhushan354 ,

Great job so far!  

May I highlight the following: 
- [x]  Well done implementing the previews TSE's requirements👏  ✔️ 
- [x] Your code is clean and well organized✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**